packages/platforms/browser/lib/routing-provider/react-router-routing-provider.ts

```diff
@@ -0,0 +1,32 @@
+import { type OnSettle } from '../on-settle'
+import { createDefaultRoutingProvider } from './default-routing-provider'
+import { stringToRegexp } from './string-to-regexp'
+
+export interface RouteObject {
+  path?: string
+  children?: RouteObject[]
+}
+
+function flattenRoutes (routes: RouteObject[], _prefix: string = ''): string[] {
+  const prefix = `${!_prefix || _prefix === '/' ? _prefix : `${_prefix}/`}`
+  return [
+    ...routes.map(route => `${prefix}${route.path || ''}`),
+    ...routes.reduce<string[]>(
+      (accum, route) => [...accum, ...(route.children ? flattenRoutes(route.children, `${prefix}${route.path}`) : [])],
+      []
+    )
+  ]
+}
+
+export const createReactRouterRoutingProvider = (onSettle: OnSettle, location: Location) => {
+  const DefaultRoutingProvider = createDefaultRoutingProvider(onSettle, location)
+
+  return class ReactRouterRoutingProvider extends DefaultRoutingProvider {
+    constructor (routes: RouteObject[], basename?: string) {
+      function resolveRoute (url: URL): string {
+        return flattenRoutes(routes).find((fullRoutePath) => url.pathname.replace(basename ?? '', '').match(stringToRegexp(fullRoutePath))) || '/'
```

Comment:
I'm not sure whether ` || '/'` is best here. Is there a better thing to do if we were unable to resolve the route? Return `url` perhaps?

Reply:
I just noticed the default route resolve defaults to `/`:

```
export const defaultRouteResolver: RouteResolver = (url: URL) => url.pathname || '/'
```

Reply:
That's just to ensure we send at least `/` even on a root URL (with no path). It's a special case because we just take the full path in that case.

In the case under discussion I would assume that `''` is an invalid route?  But perhaps we should look for the result of `.match()` returning `null` and default that to `no-route-found`.  If it is (somehow) `''` then we could convert that to `'/'`, for the same reason as in the default.